### ISSUE TITLE
Better license validation on poetry new

### ISSUE THREAD

**Issue by playpauseandstop** [sdispater/poetry#158]
**TLDR:** I am able to supply incompatible license on `poetry new` and after constantly get `Invalid license id` error on running other poetry commands.

## Steps to reproduce

1. Run `poetry new` without `--license` flag
2. On License step, enters any text, like Dummy
3. Finish `poetry new` and run any other command, like `poetry add`
4. Unexpected `Invalid license id` error happened

_Same, when passing Dummy license via `--license` command line flag._

## Technical details

- Python 3.6.5
- poetry 0.10.1
- macOS 10.11.6

## Possible solution

Here the block for reading license from user on `poetry new`,

```py
        license = self.option('license') or ''

        question = self.create_question(
            'License [<comment>{}</comment>]: '.format(license),
            default=license
        )

        license = self.ask(question)
```

To ensure user entered valid SPDX license it is needed to do something like,

```py
      while True:
          license = self.ask(question)
          try:
               license_by_id(license)
          raise ValueError:
               self.warning('Invalid license id: {}'.format(license))
          else:
               break
```

Or make refactoring in `poetry.spdx` package and create simpler way on validating license identifiers, like:

```py
def is_valid_license_id(identifier):
    load_licenses()
    return identifier.lower() in _licenses
```

**Comment by playpauseandstop** [comment]
Fixed in `poetry==0.11.1`

Thanks!
